pdns/dnsname.cc

```diff
@@ -430,29 +431,27 @@ size_t hash_value(DNSName const& d)
   return d.hash();
 }
 
-string DNSName::escapeLabel(const std::string& label)
+void DNSName::appendEscapedLabel(std::string& appendTo, const char* orig, size_t len)
 {
-  return escapeLabel(label.c_str(), label.size());
-}
-
-string DNSName::escapeLabel(const char* orig, size_t len)
-{
-  std::string ret;
   size_t pos = 0;
 
-  ret.reserve(len);
```

Comment:
Maybe this would be worth to keep `reserve()`ing enough space to append the escaped label ?

Reply:
I think that is covered by line 172. In the regular case that should be enough.

Reply:
In theory we should have reserved enough space already at the beginning of `DNSName::toString()`, but that might not be true due to escaping, I'll try to measure whether a `ret.reserve(ret.size() + len)` helps.

Reply:
> I think that is covered by line 172. In the regular case that should be enough.

Yes indeed, my bad.